### ISSUE TITLE
action cleanup

### ISSUE THREAD

**Issue by folibis** [actions/checkout#537]
The action doesn't remove a repository after workflow finished so repository just takes up space on the server. This is can be critical when using big/multiples repositories. Especially it doesn't make sense because the action deletes the repository on startup. I guess that action should provide an option to delete the repo after workflow finished.

**Comment by altjx** [comment]
Any updates on this by chance? Anyone figure out an alternative? I added a "last step cleanup" process, but seems like it breaks this action.

**Comment by Nikschavan** [comment]
Is there any alternative available for this at least? this is adding maintenance burden on the self-hosted runners where previously cloned repos keep taking up space

**Comment by anthonysidesap** [comment]
it is cleaned up , anthony j sides

**Comment by Nikschavan** [comment]
I noticed that, in my case most of the space was taken by `_temp` directory which was not getting automatically cleared.
It turned out that the cleanup was failing because of permissions, I removed the actions service and added it as a root user - https://github.com/actions/runner/issues/529#issuecomment-642016383 which fixed the issue for me.

**Comment by gavintg** [comment]
Yeah, this is a thing, would prefer a flag for `git clean -x -f -d`

**Comment by NorseGaud** [comment]
clean: true exists now
